### ISSUE TITLE
Validator Attestation Logs Appear Late

### ISSUE THREAD

**Issue by nisdas** [prysmaticlabs/prysm#4702]
@handelaar2 brought this up offline. 
```
[2020-02-01 09:04:48] DEBUG validator: No active roles, doing nothing slot=161124
[2020-02-01 09:05:00] DEBUG validator: No active roles, doing nothing slot=161125
[2020-02-01 09:05:12] DEBUG validator: No active roles, doing nothing slot=161126
[2020-02-01 09:05:24] DEBUG validator: No active roles, doing nothing slot=161127
[2020-02-01 09:05:36] DEBUG validator: No active roles, doing nothing slot=161128
[2020-02-01 09:05:48] DEBUG validator: No active roles, doing nothing slot=161129
[2020-02-01 09:06:12] DEBUG validator: No active roles, doing nothing slot=161131
[2020-02-01 09:06:12]  INFO validator: Submitted new attestations AggregatorIndices=[] AttesterIndices=[28730] BeaconBlockRoot=0x08662673d60b CommitteeIndex=7 Slot=161130 SourceEpoch=5034 SourceRoot=0xf10d6f9db824 TargetEpoch=5035 TargetRoot=0xbc085869a98e
[2020-02-01 09:06:24] DEBUG validator: No active roles, doing nothing slot=161132
[2020-02-01 09:06:36] DEBUG validator: No active roles, doing nothing slot=161133
[2020-02-01 09:06:48] DEBUG validator: No active roles, doing nothing slot=161134
[2020-02-01 09:07:00] DEBUG validator: No active roles, doing nothing slot=161135
```
The validator attestation log only comes up when the role is received for the following slot. This makes it look like the attestation is being done later.

**Comment by terencechain** [comment]
It is always been reported at the end of the slot, it has been running like this for a few months now. It's a summary view of what during the last slot. More A feature more than a bug. With that said we can add a debug log for live event. Feel free to open an issue for enhanced debug log

**Comment by nisdas** [comment]
ex: if I attest in slot 30 , shouldn't the log show up in between slot 29 and 31 ? instead of 29, 31 and then 30

**Comment by handelaar2** [comment]
Indeed, if the log entry is reported at the end of a slot, how can the log entry of slot 161131 come before 161130?
Attached is a screen print with the validator log of the previous testnet. As you can see there, all slots are in sequence (and I guess correct timewise). So in the previous testnet it looked ok to me, but not now.

The submission of attestations (and proposer blocks) is time critical. So I would say that the log should correctly indicate when an submittal was issued (to the very second and not 12 second later). Only then an analysis can start why attestations (or blocks) 'went missing'.

<img width="1450" alt="Screenshot 2020-01-31 at 13 22 00" src="https://user-images.githubusercontent.com/28559038/73595368-1e69c000-4518-11ea-9627-1e1cb41dc483.png">

**Comment by terencechain** [comment]
I see what you guys mean now. Likely they are on separate routines. This is indeed a bug. Repened

**Comment by rauljordan** [comment]
Any update on this @terencechain ?

**Comment by terencechain** [comment]
I'm not working on this. Perhaps someone who has looked into this can comment further

**Comment by prestonvanloon** [comment]
Closing this for inactivity. If this is still an issue please reopen it. Thanks!

**Comment by handelaar2** [comment]
Now that is funny Preston: "Closing this for inactivity". 
This means all issues in the end can be closed by not looking at them. Brilliant.

<img width="1006" alt="2020-03-31_08-12-37" src="https://user-images.githubusercontent.com/28559038/77993310-2e3f2e00-7328-11ea-91c6-826f8551fcf7.png">

All I want is to ensure/check that my validator submits attestations in time (to the second). For that, I cannot trust the log right now. 

But it is ok. I understand.

**Comment by prestonvanloon** [comment]
@handelaar2 Thanks for the feedback. I understand this is an important issue so I have reopened it. 

I believe this is just a cosmetic logging issue, but seems worth looking into. I think we may be launching too many goroutines unnecessarily and the log statement doesn't wait for these routines to finish. I'll submit a potential fix now.

**Comment by prestonvanloon** [comment]
@handelaar2 potential fix is https://github.com/prysmaticlabs/prysm/pull/5328, let me know if that works for you. Thanks

**Comment by rauljordan** [comment]
Hi @handelaar2 following up on this, thank you

**Comment by handelaar2** [comment]
With fix #5328, the issue has indeed been solved. 
Validator logs now appear in sequence.

<img width="1310" alt="2020-04-15_07-57-02" src="https://user-images.githubusercontent.com/28559038/79303502-55ccf380-7eef-11ea-843d-9a99579cd5ce.png">

Issue can be closed. Thanks.

**Comment by prestonvanloon** [comment]
Thanks for confirming @handelaar2
